src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -0,0 +1,151 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import subprocess
+from typing import TYPE_CHECKING, Union
+
+import numpy as np
+
+from ..utils import logging
+from .base import Pipeline
+
+
+if TYPE_CHECKING:
+    from ...feature_extraction_sequence_utils import SequenceFeatureExtractor
+
+logger = logging.get_logger(__name__)
+
+
+def ffmpeg_read(bpayload: bytes, sampling_rate: int) -> np.array:
+    """
+    Helper function to read an audio file through ffmpeg.
+    """
+    ar = f"{sampling_rate}"
+    ac = "1"
+    format_for_conversion = "f32le"
+    ffmpeg_command = [
+        "ffmpeg",
+        "-i",
+        "pipe:0",
+        "-ac",
+        ac,
+        "-ar",
+        ar,
+        "-f",
+        format_for_conversion,
+        "-hide_banner",
+        "-loglevel",
+        "quiet",
+        "pipe:1",
+    ]
+
+    try:
+        ffmpeg_process = subprocess.Popen(ffmpeg_command, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    except FileNotFoundError:
+        raise ValueError("ffmpeg was not found but is required to load audio files from filename")
+    output_stream = ffmpeg_process.communicate(bpayload)
+    out_bytes = output_stream[0]
+
+    audio = np.frombuffer(out_bytes, np.float32)
+    if audio.shape[0] == 0:
+        raise ValueError("Malformed soundfile")
+    return audio
+
+
+class AutomaticSpeechRecognitionPipeline(Pipeline):
+    """
+    Pipeline that aims at extracting spoken text contained within some audio.
+
+    The input can be either a raw waveform or a audio file. In case of the audio file, ffmpeg should be installed for
+    the read function requires it (to support many formats).
+    """
+
+    def __init__(self, feature_extractor: "SequenceFeatureExtractor", *args, **kwargs):
+        """
+        Arguments:
+            model (:obj:`~transformers.PreTrainedModel` or :obj:`~transformers.TFPreTrainedModel`):
+                The model that will be used by the pipeline to make predictions. This needs to be a model inheriting
+                from :class:`~transformers.PreTrainedModel` for PyTorch and :class:`~transformers.TFPreTrainedModel`
+                for TensorFlow.
+            tokenizer (:obj:`~transformers.PreTrainedTokenizer`):
+                The tokenizer that will be used by the pipeline to encode data for the model. This object inherits from
+                :class:`~transformers.PreTrainedTokenizer`.
+            feature_extractor (:obj:`~transformers.SequenceFeatureExtractor`):
```

Comment:
maybe put this one first since it's also first in the function signature?

Reply:
Ok.